server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -267,4 +277,154 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
+        clusterWeightRequest.attributeName("zone");
+        clusterWeightRequest.setWRRWeight(weights);
+
+        transportService.sendRequest(
+            transportService.getLocalNode(),
+            ClusterPutWRRWeightsAction.NAME,
+            clusterWeightRequest,
+            new TransportResponseHandler<ClusterPutWRRWeightsResponse>() {
+                @Override
+                public void handleResponse(ClusterPutWRRWeightsResponse response) {
+                    logger.info("Weights are successfully set.");
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    // Logging warn message on failure. Should we do Retry? If weights are not set should we fail?
+                    logger.warn("Exception occurred while setting weights.Exception Messages - [{}]", exp.unwrapCause().getMessage());
+                }
+
+                @Override
+                public String executor() {
+                    return ThreadPool.Names.SAME;
+                }
+
+                @Override
+                public ClusterPutWRRWeightsResponse read(StreamInput in) throws IOException {
+                    return new ClusterPutWRRWeightsResponse(in);
+                }
+            }
+        );
+    }
+
+    void checkHttpStatsForDecommissionedNodes(
+        Set<DiscoveryNode> decommissionedNodes,
+        String reason,
+        TimeValue timeout,
+        TimeValue timeoutForNodeDecommission,
+        ActionListener<Void> listener
+    ) {
+
+        if (timeoutForNodeDecommission.getSeconds() > 0) {
+            // Wait for timeout to happen. Log the active connection before decommissioning of nodes.
+            scheduleDecommissionNodesRequestCheck(decommissionedNodes, reason, timeout, listener, timeoutForNodeDecommission);
+        } else {
+            getActiveRequestCountOnDecommissionNodes(decommissionedNodes);
+            removeDecommissionedNodes(decommissionedNodes, reason, timeout, listener);
+        }
+    }
+
+    private void logActiveConnections(NodesStatsResponse nodesStatsResponse) {
+        Map<String, Long> nodeActiveConnectionMap = new HashMap<>();
+        List<NodeStats> responseNodes = nodesStatsResponse.getNodes();
+        for (int i = 0; i < responseNodes.size(); i++) {
+            HttpStats httpStats = responseNodes.get(i).getHttp();
+            DiscoveryNode node = responseNodes.get(i).getNode();
+            nodeActiveConnectionMap.put(node.getId(), httpStats.getServerOpen());
+        }
+        logger.info("Decommissioning node with connections : [{}]", nodeActiveConnectionMap);
+    }
+
+    private void scheduleDecommissionNodesRequestCheck(
+        Set<DiscoveryNode> decommissionedNodes,
+        String reason,
+        TimeValue timeout,
+        ActionListener<Void> nodesRemovedListener,
+        TimeValue timeoutForNodeDecommission
```

Comment:
nit : `timeoutForNodeDraining` rather than timeoutForNodeDecommission ? Decommission is whole process , of which node removal is part .

Reply:
Done.